core/services/gateway/handlers/vault/service.go

```
@@ -151,115 +154,89 @@ func (s *service) HandleJSONRPCUserMessage(ctx context.Context, jsonRequest json
 			RawResponse: s.codec.EncodeNewErrorResponse(
-				jsonRequest.ID,
-				api.ToJSONRPCErrorCode(api.InvalidParamsError),
-				fmt.Sprintf("Failed to parse request: %v", err),
+				resp.ID,
+				api.ToJSONRPCErrorCode(api.NodeReponseEncodingError),
+				fmt.Sprintf("Failed to marshal response: %v", err),
 				nil,
 			),
-			ErrorCode: api.InvalidParamsError,
-		}, callbackCh)
+			ErrorCode: api.NodeReponseEncodingError,
+		}, pendingRequest.callbackCh)
+	}
+	responseObj := gw_handlers.UserCallbackPayload{
+		RawResponse: rawResponse,
+		ErrorCode:   api.NoError,
 	}
+	// END OF DUMMY RESPONSE
 
-	// Validate request
-	if req.ID == "" {
+	// TODO: Remove the pending request from the map
+
+	s.lggr.Infof("Processed response for request %s from node %s", resp.ID, nodeAddr)
+
+	return s.sendResponse(ctx, responseObj, pendingRequest.callbackCh)
+}
+
+func (s *service) handleSecretsCreate(ctx context.Context, jsonRequest jsonrpc.Request, callbackCh chan<- gw_handlers.UserCallbackPayload) error {
+
+	var req SecretsCreateRequest
+	if err := json.Unmarshal(jsonRequest.Params, &req); err != nil {
 		return s.sendResponse(ctx, gw_handlers.UserCallbackPayload{
 			RawResponse: s.codec.EncodeNewErrorResponse(
 				jsonRequest.ID,
 				api.ToJSONRPCErrorCode(api.InvalidParamsError),
-				"Secret ID cannot be empty",
+				fmt.Sprintf("Failed to parse request: %v", err),
 				nil,
 			),
 			ErrorCode: api.InvalidParamsError,
 		}, callbackCh)
 	}
 
-	// Store secret
-	s.storeMu.Lock()
-	defer s.storeMu.Unlock()
-
-	// Extract sender from request metadata (this would come from JWT or other auth)
-	senderAddr := "default" // In a real implementation, extract from authenticated context
-	if s.secretsStore[senderAddr] == nil {
-		s.secretsStore[senderAddr] = make(map[string]SecretEntry)
-	}
-
-	// Check if secret already exists
-	if _, exists := s.secretsStore[senderAddr][req.ID]; exists {
+	// Validate request
+	if req.ID == "" || req.Value == "" {
```

Should we also atleast validate that req.Auth != ""?
If no auth provided, we should fail fast here right?

---

Auth will be a follow-up